src/domain/swap/index.ts

```diff
@@ -0,0 +1,24 @@
+export * from "./swap-out-checker"
+
+export const SwapProvider = {
```

Comment:
I'm not sure that details of the swap provider implementation should be in domain.